### PR TITLE
[xcode11.5] Bump to Xcode 11.5 Final and include provisionator changes

### PR DESCRIPTION
**Pull request by dalexsoto**
Taken from @rolfbjarne idea [here](https://github.com/xamarin/xamarin-macios/pull/8622#pullrequestreview-415058469):

> I propose a different solution to finding the right thing to ask the provisionator to install: loop through all the XreItems, looking for the one whose Url matches our XCODE_URL.

With a little twist, we only need to match xip name since that is the only thing that changes when we bump and also provisionator does not use the storage uri.

### CODE REVIEW

File: tools/devops/device-tests-provisioning.csx.in [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+		xcodeitem = item;
+		found = true;
+		break;
+	}
+
+	if (!found)
+		throw new Xamarin.Provisioning.ProvisioningException ($"'{xipFileName}' not found in 'XreItem' members.");
+
+	return xcodeitem;
```
Comment (mandel-macaque) [inline tools/devops/device-tests-provisioning.csx.in:53]: I don't think we nee the found and the break:
```suggestion
	foreach (var item in items) {
		var attr = xreItemType
			.GetField (item.ToString ())
			.GetCustomAttributes (itemAttrType, false)
			?.Cast<Xamarin.Provisioning.Model.ItemAttribute> ()
			.FirstOrDefault (a => a.Uri.EndsWith (xipFileName, StringComparison.Ordinal));

		if (attr == null)
			continue;

		return item;
	}

	throw new Xamarin.Provisioning.ProvisioningException ($"'{xipFileName}' not found in 'XreItem' members.");

```

Comment (dalexsoto) [reply tools/devops/device-tests-provisioning.csx.in]: Fixed, thanks I got rid of the variables too :)

**Comment by dalexsoto** [comment]
Known failure: HE0038

**Comment by dalexsoto** [comment]
Build

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-8634/1/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-8634/1/display/redirect) (on internal Jenkins) failed in stage(s) 'Checkout' 🔥 : hudson.AbortException: script returned exit code 128

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-8634/2/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-8634/2/display/redirect) (on internal Jenkins) failed in stage(s) 'Checkout' 🔥 : hudson.AbortException: script returned exit code 128

**Pull request merged**